hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -806,6 +878,225 @@ public Bytes getFailureResultFor(final ResponseCodeEnum status) {
 		}
 	}
 
+	/**
+	 * Executes the logic of creating a token from {@link HTSPrecompiledContract}.
+	 *
+	 * <p>When a token create call is received, the execution can follow one of the following 4 scenarios:</p>
+	 * <ol>
+	 *     <li>The calling smart contract has sent a corrupt input to the {@link HTSPrecompiledContract} (e.g.
+	 *     passing random bytes through the {@code encode()} Solidity method), which cannot be decoded to a valid
+	 *     {@link TokenCreateWrapper}.
+	 *     		<ul>
+	 *     		 	<li><b>result</b> - (TBD) the {@link DecodingFacade} throws an exception and null is returned
+	 *     		 	from the {@link HTSPrecompiledContract}</li>
+	 *     		 	<li><b>gas cost</b> - TBD</li>
+	 *     		</ul>
+	 *     	</li>
+	 *     <li>The decoding succeeds, we create a valid {@link TokenCreateWrapper}, but we cannot translate it to a
+	 *     valid token create {@link TransactionBody}. This comes from <b>difference in the design of the Solidity
+	 *     function interface and the HAPI API (protobufs)</b>
+	 *     		<ul>
+  	 *     		 	<li><b>result</b> - {@link MessageFrame}'s revertReason is set to the specific error and null is
+	 *     		 	returned from the {@link HTSPrecompiledContract}</li>
+	 * 	    		<li><b>gas cost</b> - TBD</li>
```

Comment:
Again, the gas cost here will be the value returned from dynamicProperties.htsDefaultGasCost()